### PR TITLE
Include python Stream in Step module

### PR DESCRIPTION
**Pull request by DumouchelP**
* **Please check if the PR fulfills these requirements**
- [x] The commit messages follows our guidelines (explicit, in english)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] CHANGELOG.md has been updated

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
- [ ] Bug fix:
- [x] New features:
- [ ] Documentation
- [ ] Optimization

* **Other information**:

### CODE REVIEW

**Review by AxelRinghausen — DISMISSED** [review]

**Review by masfaraud — CHANGES_REQUESTED** [review]
Create a separate method for stream

File: volmdlr/step.py [hunk 1]
```diff
@@ -242,22 +243,42 @@ def simplify(self, new_name):
 
 
 class Step:
-    def __init__(self, stepfile):
+    def __init__(self, stepfile: str = None, stream: BinaryIO = None):
```
Comment (masfaraud) [inline volmdlr/step.py:246]: Weird to have double args in init. I suggest init with lines (list[str]) and have classmethods to read from stream and from file

File: volmdlr/step.py [hunk 2]
```diff
@@ -242,22 +243,42 @@ def simplify(self, new_name):
...
+        if stepfile is not None:
+            self.functions, self.all_connections = self.read_stepfile()
+        if stream is not None:
+            self.functions, self.all_connections = self.read_stream()
+        if stepfile is None and stream is None:
+            raise KeyError('choose between stepfile or stream')
         self.upd_graph = False
 
-    def read_functions(self):
+    def read_stream(self):
```
Comment (masfaraud) [inline volmdlr/step.py:258]: have stream in param not in self

**Review by masfaraud — CHANGES_REQUESTED** [review]

**Review by DumouchelP — COMMENTED** [review]
2 new class_method to instantiate Step

**Comment by masfaraud** [comment]
Embeded in another PR

**Pull request closed without merging**